gibberish_client/src/components/Gameroom/GameroomPage.js

```diff
@@ -14,28 +14,61 @@ class GameroomPage extends React.Component {
 			roomId: props.match.params.id,
 			gamestate: gamestates.GAME_WAITING,
 			currentRound: 1,
-			maxRounds: 5,
-			players: [
-				{name: this.props.nickname, points: 0},
-				{name: "gibberish gary", points: 0},
-				{name: "nonsensical nick", points: 0},
-				{name: "blah blah brenda ", points: 0},
-			],
+			players: [],
 			roundScores: [],
+			qna: [],
 			currentQuestion: '',
 			currentAnswer: '', 
 			userAnswer: '',
 			timeRemaining: 0,
-			helpText: ''
+			helpText: '',
 		}
 	}
 
 	componentDidMount() {
 		this.getQuestion();
+		const {roomId} = this.state
+		this.timer = setInterval(() => this.getRoomData(roomId), 1000)
 	}
 
-	setTimeRemaining = (time) => {
-		this.setState({timeRemaining: time})
+	componentWillUnmount() {
+		this.timer = null
+	}
+
+	getRoomData(roomId) {
+		API.get(`/room/${roomId}`)
+			.then(res => {
+				this.parseRoomData(res.data)
+			})
+	}
+
+	parseRoomData(data) {
+		const {roomId, gameState, currentRound, players, timer, qna} = data 
+		let timeleft
+		if(gameState == gamestates.ROUND_LOADING) {
+			timeleft = 3-timer
+		} else if(gameState == gamestates.ROUND_ONGOING) {
+			timeleft = 10 - timer
+		} else if(gameState == gamestates.ROUND_ENDED) {
+			timeleft = 5 - timer
```

Comment:
please variables these magic numbers (3, 10, 5) as variables so that it's easy to understand what they represent.

Reply:
the magic numbers have been assigned to constants in the latest commits 👍